### ISSUE TITLE
Span and ReadonlySpan only show up in completions with unopened namespaces

### ISSUE THREAD

**Issue by cartermp** [dotnet/fsharp#5858]
1. Create a new .NET Core console app targeting .NET Core 2.1.
2. Ensure that Tools > Options > Text Editor > F# > IntelliSense > Show symbols in unopened namespaces is off

![image](https://user-images.githubusercontent.com/6309070/47930125-41767180-de88-11e8-961f-d772c0a38292.png)

Attempt to get a `Span` or `ReadOnlySpan` from a completion list and notice that it doesn't show up:

![image](https://user-images.githubusercontent.com/6309070/47930175-5f43d680-de88-11e8-8b5e-d0a1ce03f4bd.png)
![image](https://user-images.githubusercontent.com/6309070/47930182-666ae480-de88-11e8-8a69-e8983f1967d3.png)

And notice that if you turn the feature back on, they show up, but only after opening `System` (even when it's open already):

![image](https://user-images.githubusercontent.com/6309070/47930249-96b28300-de88-11e8-921c-704461d4ffa3.png)

**Comment by TIHan** [comment]
Yes, we are not doing the check for Obsolete attribute correctly, as we also have to match the specific text that is inside of it. 

Technically, we can fix this specific bug in 2 lines of code in NameResolution, but I think there is a better overall design to handle this. Basically, whenever we ask for a list of attributes for a ByrefLike type, we should not include the Obsolete attribute with the specific text in the list; that way we don't need to special case Obsolete and ByrefLike in multiple places in our codebase. However, upon trying to figure out how to do this, we might have an issue with attributes not giving us all the data we need. I will look at this again Monday.

**Comment by dsyme** [comment]
This was fixed by #5866

**Comment by dsyme** [comment]
@cartermp Was this fixed by #5866?

**Comment by cartermp** [comment]
Yes, this would have been resolved by that. But It appears that something was overcomplicated.

Not sure I agree with low severity here. This means that these types (and another others marked this way) are not discoverable via IntelliSense.

**Comment by brianrourkeboll** [comment]
I still think this is worth fixing.

**Comment by T-Gro** [comment]
Yes, the `[Obsolete]` attribute should be thrown away at earliest import point - for the cases where it only exists for older compilers.
(which is/was a strange misuse, considering `modreq` exists)

**Comment by T-Gro** [comment]
Related test:

https://github.com/dotnet/fsharp/blob/d04f46f4d3c2db8ee8dcd050dc56c6987f445e60/tests/FSharp.Compiler.Service.Tests/CompletionTests.fs#L372-L379

The `Span appears in completion and is not marked obsolete` test verifies that `Span` appears in the completion list when typing `System.Sp` with the cursor positioned after `Sp`.
